modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogParamsValidationUtils.java

```diff
@@ -95,6 +105,83 @@ static void validateRenameZoneParams(RenameZoneParams params) {
         validateZoneName(params.newZoneName(), "Missing new zone name");
     }
 
+    static void validateDropTableParams(DropTableParams params) {
+        validateCommonTableParams(params);
+    }
+
+    static void validateCreateTableParams(CreateTableParams params) {
+        validateCommonTableParams(params);
+
+        List<String> columnNames = Objects.<List<ColumnParams>>requireNonNullElse(params.columns(), List.of()).stream()
+                .peek(CatalogParamsValidationUtils::validateColumnParams)
+                .map(ColumnParams::name)
+                .collect(toList());
+
+        validateColumns(
+                columnNames,
+                Table.TABLE_DEFINITION_ERR,
+                "Columns not specified",
+                "Duplicate columns are present: {}"
+        );
+
+        validateColumns(
+                params.primaryKeyColumns(),
+                Table.TABLE_DEFINITION_ERR,
+                "Primary key columns not specified",
+                "Duplicate primary key columns are present: {}"
+        );
+
+        validateColumnsContainsInAnotherColumns(
+                params.primaryKeyColumns(),
+                columnNames,
+                Table.TABLE_DEFINITION_ERR,
+                "Primary key columns missing in columns: {}"
+        );
+
+        validateColumns(
+                params.colocationColumns(),
+                Table.TABLE_DEFINITION_ERR,
+                "Colocation columns not specified",
+                "Duplicate colocation columns are present: {}"
+        );
```

Comment:
AFAIK, we use PK columns if colocation columns were not set.
So, we have to check on duplicates only.

Reply:
In `org.apache.ignite.internal.schema.configuration.TableValidatorImpl#validate`L96 it is expected that it is always exists, I repeated the logic here.
As in the current implementation, see `org.apache.ignite.internal.sql.engine.exec.ddl.DdlCommandHandler#handleCreateTable`L279, and now in `org.apache.ignite.internal.sql.engine.exec.ddl.DdlToCatalogCommandConverter#convert(org.apache.ignite.internal.sql.engine.prepare.ddl.CreateTableCommand)`L73 in current PR.

Reply:
You refer to a legacy code.

`org.apache.ignite.internal.catalog.descriptors.CatalogTableDescriptor`L77 still ok with `colocationColumns == null`.
I would add any logic in parameters conversion.

Reply:
Okay, I'll fix this.